bin/i18n/curriculum_sync_utils.rb

```diff
@@ -0,0 +1,204 @@
+require 'cdo/languages'
+
+require_relative 'curriculum_sync_utils/serializers'
+
+# This module contains all i18n sync logic specific to the content
+# imported into Code Studio from CurriculumBuilder; Lessons, Activities,
+# Resources, Vocabularies, Objectives, etc.
+module CurriculumSyncUtils
+  def self.sync_in
+    puts "Sync in curriculum content"
+    sync_in_serialize
+    # TODO: redaction
+  end
+
+  def self.sync_out
+    puts "Sync out curriculum content"
+    # TODO: restoration
+    sync_out_reorganize
+  end
+
+  # Helper method to get the desired destination subdirectory of the given
+  # script for the sync in. Note this may be a nested directory like "2021/csf"
+  def self.get_script_subdirectory(script)
+    # special-case Hour of Code scripts.
+    return "Hour of Code" if ScriptConstants.script_in_category?(:hoc, script.name)
+
+    # catchall for scripts without courses
+    return 'other' unless script.get_course_version.present?
+
+    # special-case CSF; we want to group all CSF courses together, even though
+    # they all have different course offerings.
+    return File.join(script.get_course_version.key, 'csf') if script.csf?
+
+    # base case
+    return File.join(script.get_course_version.key, script.get_course_version.course_offering.key)
+  end
+
+  # Serialize all curriculum data to the I18N source directory.
+  #
+  # Relies heavily on the custom serializer logic defined in
+  # curriculum_sync_utils/serializers to generate a hash of results with
+  # translator-readable keys, rather than the basic array that is produced by
+  # default.
+  def self.sync_in_serialize
+    Script.all.each do |script|
+      # TODO: what else do we want to consider when deciding what to sync?
+      next unless script.is_migrated?
```

Comment:
we are soon planning to migrate all scripts, however any scripts which are not part of CSF / CSD / CSP 2021 will not have any lesson plan content. do you know whether you will want these scripts to be translated in this new way or the old way?

as time progresses, we'll add lesson plan content for some additional scripts, starting with HOC scripts.

Reply:
none of this work involves translating scripts themselves, just their lesson plan content

Reply:
ok -- what kind of input (if any) are you looking for to help answer this question? "what else do we want to consider when deciding what to sync?"

Reply:
oh, this is just a note left to myself. One of the "pile of details" things I mentioned in standup today :upside_down_face:

Reply:
What happens when we do start marking scripts without lesson plan content as migrated here? I realize this is likely part of the details you're sorting out but as it is I think it would just produce an empty hash?